File: app/models/airport.py
from sqlalchemy import Column, String, Float
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship
import uuid
from app.db.base import Base

class Airport(Base):
    __tablename__ = "airports"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    icao_code = Column(String(4), unique=True, nullable=False)  # ICAO code (e.g., LTBA)
    iata_code = Column(String(3), unique=True, nullable=True)   # IATA code (e.g., IST)
    name = Column(String, nullable=False)                       # Airport name
    city = Column(String, nullable=False)                       # City
    country = Column(String, nullable=False)                    # Country
    
    # Location
    latitude = Column(Float, nullable=False)                    # Latitude in decimal degrees
    longitude = Column(Float, nullable=False)                   # Longitude in decimal degrees
    elevation = Column(Float, nullable=True)                    # Elevation in meters

    # Relationships
    departing_flights = relationship("Aircraft", foreign_keys="[Aircraft.departure_airport_id]", back_populates="departure_airport")
    arriving_flights = relationship("Aircraft", foreign_keys="[Aircraft.destination_airport_id]", back_populates="destination_airport") 

File: backend/models/airport.go
package models

import (
	"time"
)

type Airport struct {
	ID               int           `gorm:"primaryKey;column:id" json:"id"`
	Name             string        `gorm:"column:name;not null" json:"name"`
	IcaoCode         string        `gorm:"column:icao_code;unique" json:"icao_code"`
	IataCode         string        `gorm:"column:iata_code;unique" json:"iata_code"`
	Latitude         float64       `gorm:"column:latitude;not null" json:"latitude"`
	Longitude        float64       `gorm:"column:longitude;not null" json:"longitude"`
	Elevation        float64       `gorm:"column:elevation" json:"elevation"`
	City             string        `gorm:"column:city" json:"city"`
	Country          string        `gorm:"column:country" json:"country"`
	AirStatusCelcius int           `gorm:"column:air_status_celcius" json:"air_status_celcius"`
	RunwayLength     float64       `gorm:"column:runway_length" json:"runway_length"`
	IsMilitary       bool          `gorm:"column:is_military;default:false" json:"is_military"`
	Status           bool          `gorm:"column:status;default:true" json:"status"`
	WeatherStatus    WeatherStatus `gorm:"column:weather_status" json:"weather_status"`
	CreatedAt        time.Time     `gorm:"column:created_at;default:current_timestamp" json:"created_at"`
	UpdatedAt        time.Time     `gorm:"column:updated_at;default:current_timestamp" json:"updated_at"`
	
	// Relationships
	OriginFlightPlans []FlightPlan `gorm:"foreignKey:DepartureAirportID" json:"-"`
	ArrivalFlightPlans []FlightPlan `gorm:"foreignKey:ArrivalAirportID" json:"-"`
	AlternateFlightPlans []FlightPlan `gorm:"foreignKey:AlternateAirportID" json:"-"`
	WeatherConditions []WeatherCondition `gorm:"foreignKey:AirportID" json:"-"`
	BasedAircrafts []Aircraft `gorm:"foreignKey:CurrentLocation" json:"-"`
	AircraftHistory []AircraftHistory `gorm:"foreignKey:Location" json:"-"`
}

func (Airport) TableName() string {
	return "airports"
} 